flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/AbstractFlinkService.java

```diff
@@ -182,7 +184,13 @@ public JobID submitJobToSessionCluster(
             throws Exception {
         // we generate jobID in advance to help deduplicate job submission.
         var jobID = FlinkUtils.generateSessionJobFixedJobID(meta);
-        runJar(spec.getJob(), jobID, uploadJar(meta, spec, conf), conf, savepoint);
+        Configuration runtimeConfig = removeOperatorConfigs(conf);
+        runJar(
+                spec.getJob(),
+                jobID,
+                uploadJar(meta, spec, runtimeConfig),
+                runtimeConfig,
+                savepoint);
```

Comment:
I looked at the code and I think for sessionJobs we should not remove any configs. When we submit the jar through the rest api the configuration is not passed. 

Removing operator configs here can actually break some artifact fetcher mechanisms within the operator here.

We don't have very good test coverage it seems that would catch this problem. I will open some tickets

Reply:
Thanks for the suggestion. Revised acccordingly.